plugins/handler/logs/main.go

```diff
@@ -0,0 +1,185 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"time"
+	"bytes"
+	"strconv"
+	"encoding/json"
+
+	"github.com/infrawatch/sg-core/pkg/bus"
+	"github.com/infrawatch/sg-core/pkg/config"
+	"github.com/infrawatch/sg-core/pkg/data"
+	"github.com/infrawatch/sg-core/pkg/handler"
+	"github.com/infrawatch/sg-core/plugins/handler/logs/pkg/lib"
+)
+
+type SyslogSeverity int
+
+const (
+	EMERGENCY SyslogSeverity = iota
+	ALERT
+	CRITICAL
+	ERROR
+	WARNING
+	NOTICE
+	INFORMATIONAL
+	DEBUG
+	UNKNOWN
+)
+
+func (rs SyslogSeverity) toEventSeverity() data.EventSeverity {
+	return []data.EventSeverity{data.CRITICAL,
+	                            data.CRITICAL,
+	                            data.CRITICAL,
+	                            data.CRITICAL,
+	                            data.WARNING,
+	                            data.INFO,
+	                            data.INFO,
+	                            data.INFO,
+	                            data.UNKNOWN,
+	                           }[rs]
+}
+
+type logConfig struct {
+	MessageField   string `validate:"required"`
+	TimestampField string `validate:"required"`
+	HostnameField  string `validate:"required"`
+	SeverityField  string
+}
+
+type logHandler struct {
+	totalLogsReceived uint64
+	config            logConfig
+}
+
+func (l *logHandler) parse(log []byte) (data.Event, error) {
+	parsedLog := data.Event{}
+	logFields := make(map[string]interface{})
+	err := json.Unmarshal(log, &logFields)
+	if err != nil {
+		return parsedLog, err
+	}
+
+	msg, ok := logFields[l.config.MessageField].(string)
+	if !ok {
+		return parsedLog, fmt.Errorf("Unable to find a log message under field called: %s", l.config.MessageField)
+	}
+
+	severity := UNKNOWN
+	severitystring, ok := logFields[l.config.SeverityField].(string)
+	if ok {
+		s, err := strconv.Atoi(severitystring)
+		if err == nil {
+			severity = SyslogSeverity(s)
+		}
+	}
+	eventSeverity := severity.toEventSeverity()
+
+	hostname, ok := logFields[l.config.HostnameField].(string)
+	if !ok {
+		return parsedLog, fmt.Errorf("Unable to find the hostname under field called: %s", l.config.HostnameField)
+	}
+
+	timestring, ok := logFields[l.config.TimestampField].(string)
+	if !ok {
+		return parsedLog, fmt.Errorf("Unable to find the timestamp under field called: %s", l.config.TimestampField)
+	}
+	t, err := lib.TimeFromFormat(timestring)
+	if err != nil {
+		return parsedLog, err
+	}
+
+	timestamp := float64(t.Unix())
+	year, month, day := t.Date()
+
+	index := fmt.Sprintf("logs-%s-%d-%d-%d", hostname, year, month, day)
+
+
+	// remove message and timestamp from labels (leave the rest)
+	delete(logFields, l.config.MessageField)
+	delete(logFields, l.config.TimestampField)
+
+	parsedLog = data.Event {
+		Index: index,
+		Time: timestamp,
+		Type: data.LOG,
+		Publisher: "sg-core",
```

Comment:
Does the rsyslog message contain information about what node the message came from? That would be more useful than 'sg-core' here

Reply:
Agree, use sg-core only when hostname is not there.

Reply:
That would be your hostname variable above most likely

Reply:
And make sure you're sending the full hostname for multi-cloud compatibility. We already have that fix required for libpodstats :D

Reply:
😀 calling me out, I see. In this case, it's up to rsyslog sending the fqdn though

Reply:
I didn't post blame for anyone. You took that all on your own with this comment lol

Reply:
I wasn't sure what exactly Publisher is. The hostname variable should work as long as rsyslog is configured correctly.